src/rpc/core_rpc_server.cpp

```diff
@@ -333,12 +334,32 @@ namespace cryptonote
     if (m_rpc_payment)
       m_net_server.add_idle_handler([this](){ return m_rpc_payment->on_idle(); }, 60 * 1000);
 
+    bool ssl_key_needed = rpc_config->ssl_options.auth.certificate_path.empty();
+    const auto ssl_base_path = (boost::filesystem::path{data_dir} / (restricted ? "restricted_rpc_ssl" : "rpc_ssl")).string();
+    if (ssl_key_needed && boost::filesystem::exists(ssl_base_path + ".crt"))
+    {
+      // load key from previous run, password prompted by OpenSSL
```

Comment:
That is correct. I had code to store encrypted, and its fairly easy to add. But it doesn't work if someone is launching without TTY/console. I thought about refactoring the checks for password prompt so that it stores unencrypted in such scenarios, but I wasn't sure about it.

So - change so that `if (is_tty() and !password.empty()) { encrypt ... }` ? The log will contain a reasonable error when things go awry.

Reply:
This was mostly a comment about the comment being the opposite of the code. I guess encrypted is always better all other things being equal too, but I agree no TTY is annoying so I'm fine with either here.